src/wrappers/python/PyOpenEXR.cpp

```diff
@@ -1040,9 +1040,34 @@ PyFile::write(const char* outfilename)
         
         for (auto a : P.header)
         {
-            auto name = py::str(a.first);
-            py::object second = py::cast<py::object>(a.second);
-            insertAttribute(header, name, second);
+            std::string name   = a.first.cast<std::string> ();
```

Comment:
I think these if statements are all unnecessary, since they are each cases handled by `insertAttribute()`, which also handles the various cases like conversion from tuples.

You also seem to have lost the crux of the original fix, which is the comparison of the dataWindow to the part shape.